src/Illuminate/Routing/RouteDependencyResolverTrait.php

```diff
@@ -52,14 +50,25 @@ protected function resolveClassMethodDependencies(array $parameters, $instance,
     public function resolveMethodDependencies(array $parameters, ReflectionFunctionAbstract $reflector)
     {
         $originalParameters = $parameters;
+        $parameters = array_values($parameters);
```

Comment:
`$parameters` are **associative array**, which will produce unpredictable behavior when apply to `call_user_func_array`. So I keep the numerical index here.